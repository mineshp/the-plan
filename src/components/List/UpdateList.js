import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Dropdown, Form, Grid, Header, Input } from 'semantic-ui-react';
import HeadingInput from './HeadingInput';
import './List.css';

const UpdateList = ({
    projectOptions,
    result,
    handleSubmit,
    handleChange,
    handleDropDownSelection,
    handleHeaderInputChange,
    addHeading,
    removeHeading,
    headings
}) => (
    <Container>
        <Header as="h2">Setup List</Header>
        <Form onSubmit={handleSubmit}>
            <Grid columns={1}>
                <Grid.Row>
                    <Grid.Column>
                        <Form.Field>
                            <Input
                                placeholder="List Name..."
                                className="text-box-single-col-max"
                                defaultValue={result.listName}
                                onChange={handleChange}
                            />
                        </Form.Field>
                        <Form.Field>
                            <Dropdown
                                placeholder="Projects"
                                fluid
                                multiple
                                search
                                selection
                                options={projectOptions}
                                onChange={handleDropDownSelection}
                            />
                        </Form.Field>
                    </Grid.Column>
                </Grid.Row>
            </Grid>
            <HeadingInput
                headings={headings}
                handleAddColumn={addHeading}
                handleRemoveColumn={removeHeading}
                handleHeaderInputChange={handleHeaderInputChange}
            />
            <Grid columns={1}>
                <Grid.Row>
                    <Button fluid className="formSpacing" color="teal" type="submit">Save</Button>
                </Grid.Row>
            </Grid>
        </Form>
    </Container>
);

UpdateList.propTypes = {
    projectOptions: PropTypes.arrayOf(
        PropTypes.shape({})
    ).isRequired,
    result: PropTypes.shape({}),
    handleSubmit: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleDropDownSelection: PropTypes.func.isRequired,
    handleHeaderInputChange: PropTypes.func.isRequired,
    addHeading: PropTypes.func.isRequired,
    removeHeading: PropTypes.func.isRequired,
    headings: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string
    }).isRequired).isRequired
};

UpdateList.defaultProps = {
    result: {}
};


export default UpdateList;
