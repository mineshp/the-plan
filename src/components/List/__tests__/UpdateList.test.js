import React from 'react';
import renderer from 'react-test-renderer';
import UpdateList from '../UpdateList';
import { mockHeadingsData, mockProjectDropDownOptionsData, mockNewListResultData, mockExistingListWithNoProjects } from '../../../helpers/test/testData/listData';

const mockHandleSubmit = jest.fn();
const mockHandleChange = jest.fn();
const mockHandleDropDownSelection = jest.fn();
const mockHandleHeaderInputChange = jest.fn();
const mockAddHeading = jest.fn();
const mockRemoveHeading = jest.fn();

const mockProjectOptions = mockProjectDropDownOptionsData();

const mockHeadings = mockHeadingsData();

describe('Create New List', () => {
    it('renders create new list form correctly', () => {
        const tree = renderer.create(<UpdateList
            projectOptions={mockProjectOptions}
            result={mockNewListResultData()}
            handleSubmit={mockHandleSubmit}
            handleChange={mockHandleChange}
            handleDropDownSelection={mockHandleDropDownSelection}
            handleHeaderInputChange={mockHandleHeaderInputChange}
            addHeading={mockAddHeading}
            removeHeading={mockRemoveHeading}
            headings={mockHeadings}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Update List', () => {
    const mockExistingListWithProjects = Object.assign({}, mockExistingListWithNoProjects, {
        projects: [
            {
                id: '001',
                name: 'History'
            }
        ]
    });

    it('renders existing list form correctly when no projects have been assigned', () => {
        const tree = renderer.create(<UpdateList
            projectOptions={mockProjectOptions}
            result={mockExistingListWithNoProjects()}
            handleSubmit={mockHandleSubmit}
            handleChange={mockHandleChange}
            handleDropDownSelection={mockHandleDropDownSelection}
            handleHeaderInputChange={mockHandleHeaderInputChange}
            addHeading={mockAddHeading}
            removeHeading={mockRemoveHeading}
            headings={mockHeadings}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders existing list form correctly when projects have already been assigned', () => {
        const tree = renderer.create(<UpdateList
            projectOptions={mockProjectOptions}
            result={mockExistingListWithProjects}
            handleSubmit={mockHandleSubmit}
            handleChange={mockHandleChange}
            handleDropDownSelection={mockHandleDropDownSelection}
            handleHeaderInputChange={mockHandleHeaderInputChange}
            addHeading={mockAddHeading}
            removeHeading={mockRemoveHeading}
            headings={mockHeadings}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
