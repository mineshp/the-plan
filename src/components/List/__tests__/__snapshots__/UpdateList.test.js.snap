// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create New List renders create new list form correctly 1`] = `
<div
  className="ui container"
>
  <h1
    className="ui header"
  >
    Update New Avengers
  </h1>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <div
          className="column"
        >
          <div
            className="field"
          >
            <div
              className="ui input text-box-single-col-max"
            >
              <input
                defaultValue="New Avengers"
                disabled={undefined}
                onChange={[Function]}
                placeholder="List Name..."
                tabIndex={undefined}
                type="text"
              />
            </div>
          </div>
          <div
            className="ui horizontal divider"
          >
            ASSIGN LIST TO PROJECTS
          </div>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid multiple search selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <span
                className="sizer"
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Projects
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={true}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj1
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj2
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ui divided two column grid"
      >
        <div
          className="ui horizontal divider"
        >
          ASSIGN TABLE HEADINGS
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="A"
                  disabled={undefined}
                  id="000001"
                  name="A"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="B"
                  disabled={undefined}
                  id="000002"
                  name="B"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="ui divider"
        />
        <button
          className="ui blue fluid button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="button"
        >
          Add Column
        </button>
      </div>
    </div>
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <button
          className="ui green fluid button formSpacing"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Update List renders existing list form correctly when no projects have been assigned 1`] = `
<div
  className="ui container"
>
  <h1
    className="ui header"
  >
    Update Original Avengers
  </h1>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <div
          className="column"
        >
          <div
            className="field"
          >
            <div
              className="ui input text-box-single-col-max"
            >
              <input
                defaultValue="Original Avengers"
                disabled={undefined}
                onChange={[Function]}
                placeholder="List Name..."
                tabIndex={undefined}
                type="text"
              />
            </div>
          </div>
          <div
            className="ui horizontal divider"
          >
            ASSIGN LIST TO PROJECTS
          </div>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid multiple search selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <span
                className="sizer"
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Projects
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={true}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj1
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj2
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ui divided two column grid"
      >
        <div
          className="ui horizontal divider"
        >
          ASSIGN TABLE HEADINGS
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="A"
                  disabled={undefined}
                  id="000001"
                  name="A"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="B"
                  disabled={undefined}
                  id="000002"
                  name="B"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="ui divider"
        />
        <button
          className="ui blue fluid button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="button"
        >
          Add Column
        </button>
      </div>
    </div>
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <button
          className="ui green fluid button formSpacing"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Update List renders existing list form correctly when projects have already been assigned 1`] = `
<div
  className="ui container"
>
  <h1
    className="ui header"
  >
    Setup List
  </h1>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <div
          className="column"
        >
          <div
            className="field"
          >
            <div
              className="ui input text-box-single-col-max"
            >
              <input
                defaultValue={undefined}
                disabled={undefined}
                onChange={[Function]}
                placeholder="List Name..."
                tabIndex={undefined}
                type="text"
              />
            </div>
          </div>
          <div
            className="ui horizontal divider"
          >
            ASSIGN LIST TO PROJECTS
          </div>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid multiple search selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <span
                className="sizer"
              />
              <div
                aria-live="polite"
                className="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={true}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj1
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj2
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    proj3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="ui divided two column grid"
      >
        <div
          className="ui horizontal divider"
        >
          ASSIGN TABLE HEADINGS
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="A"
                  disabled={undefined}
                  id="000001"
                  name="A"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  defaultValue="B"
                  disabled={undefined}
                  id="000002"
                  name="B"
                  onChange={[Function]}
                  placeholder="Header Name..."
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <button
              className="ui pink icon left floated left labeled button"
              disabled={undefined}
              id="000002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="button"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Remove
            </button>
          </div>
        </div>
        <div
          className="ui divider"
        />
        <button
          className="ui blue fluid button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="button"
        >
          Add Column
        </button>
      </div>
    </div>
    <div
      className="ui one column grid"
    >
      <div
        className="row"
      >
        <button
          className="ui green fluid button formSpacing"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;
